numpy/lib/function_base.py

```diff
@@ -4277,7 +4277,20 @@ def meshgrid(*xi, copy=True, sparse=False, indexing='xy'):
 
         .. versionadded:: 1.7.0
     sparse : bool, optional
-        If True a sparse grid is returned in order to conserve memory.
+        If True the shape of the returned coordinate array for dimension *i*
+        is reduced from ``(N1, ..., Ni, ... Nn)`` to
+        ``(1, ..., 1, Ni, 1, ..., 1)``.  These spare coordinate grids can
+        conserve memory and computation time.  When all coordinates are used
+        in an expression, broadcasting still leads to a fully-dimensonal
+        result array.
+
+        Example: In the second example below with a (101, 101) shaped grid,
+        the total size of the coordinate arrays ``xx`` and ``yy`` is reduced
+        from 2*101*101 = 20.402 to 101+101 = 202 values. Likewise, the
+        expression ``xx**2 + yy**2`` takes 20.401 square operations with full
+        coordinates but only 202 square operations with sparse coordinates,
+        while the resulting is the same.
+
```

Comment:
I think this explanatory text would fit better directly in the examples section. That way some of these points could be illustrated directly, e.g. comparing `xx.shape` w/ both `sparse=True/False`